src/main/java/tsunami/security/scanner/utilities/App.java

```diff
@@ -16,12 +16,180 @@
+        JCommander helloCmd = JCommander.newBuilder()
+                .addObject(jArgs)
+                .build();
+        helloCmd.parse(args);
+
+        String appName = jArgs.getName();
+        String appVersion = jArgs.getVersion();
+
+        System.out.println("Hello " + appName + " " + appVersion);
+
+        /* Logic to get the relative application config */
+        String configPath = System.getProperty("user.dir") + "/application/" + appName + "/";
+        File file = new File(configPath);
+        if (file.isDirectory()) {
+            try {
+                File[] files = file.listFiles();
+                for (File f : files) {
+                    if (!f.isDirectory())
+                        /* TODO: parse all config files under a certain app name to Kubernetes Object here, maybe use index to represent each config file*/
+                        System.out.println(f);
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+
+
+        /* Creation of Pod and Service without config files
+        V1Pod pod =
+                new V1PodBuilder()
+                        .withNewMetadata()
+                        .withName("apod")
+                        .endMetadata()
+                        .withNewSpec()
+                        .addNewContainer()
+                        .withName("www")
+                        .withImage("nginx")
+                        .withNewResources()
+                        .withLimits(new HashMap<>())
+                        .endResources()
+                        .endContainer()
+                        .endSpec()
+                        .build();
+        System.out.println(Yaml.dump(pod));
+
+        Map<String,String> m = new HashMap<>();
+        Map<String,String> m2 = new HashMap<>();
+        V1Service svc =
+                new V1ServiceBuilder()
+                        .withNewMetadata()
+                        .withName("wordpress")
+                        .withLabels(m)
+                        .addToLabels("app", "wordpress")
+                        .endMetadata()
+                        .withNewSpec()
+                        .addNewPort()
+                        .withPort(80)
+                        .endPort()
+                        .withSelector(m2)
+                        .addToSelector("app","wordpress")
+                        .addToSelector("tier", "frontend")
+                        .withType("LoadBalancer")
+                        .endSpec()
+                        .build();
+        System.out.println(Yaml.dump(svc));
+        */
+
+
+        // file path to KubeConfig
+        String kubeConfigPath = System.getProperty("user.home") + "/.kube/config";
+
+        // loading the out-of-cluster config, a kubeconfig from file-system
+        ApiClient client =
+                ClientBuilder.kubeconfig(KubeConfig.loadKubeConfig(new FileReader(kubeConfigPath))).build();
```

Comment:
I think you can simply use `ApiClient client = Config.defaultClient();` to get an instance of `ApiClient`. `defaultClient` will work both locally and within a cluster. See comment [here](https://github.com/kubernetes-client/java/blob/cbc1d385dfa02da38e07e4b22498971b7b464680/util/src/main/java/io/kubernetes/client/util/Config.java#L103-L108).

Reply:
done